models/issue_milestone.go

```diff
@@ -236,7 +236,11 @@ func GetMilestonesByRepoID(repoID int64, state api.StateType) (MilestoneList, er
 		sess = sess.And("is_closed = ?", false)
 	}
 
-	miles := make([]*Milestone, 0, 10)
+	if listOptions.Page != 0 {
+		sess = listOptions.setSessionPagination(sess)
+	}
+
+	var miles []*Milestone
```

Comment:
Sorry, That's not what I meant. You should still preallocate the array - it's a lot more efficient to do so. Just preallocate it to the size of the page.

Reply:
https://github.com/go-gitea/gitea/pull/9452/commits/1d016ade85bdccabea9522c9e706b849b2f1f159

Improves memory allocations on multiple functions